packages/main/src/MessageStripContext.js

```diff
@@ -0,0 +1,45 @@
+import MessageStripType from "./types/MessageStripType";
+
+class MessageStripTemplateContext {
+	static calculate(state) {
+		return {
+			ctr: state,
+			classes: {
+				type: MessageStripTemplateContext.getTypeClasses(state),
+				label: MessageStripTemplateContext.getLabelClasses(state),
+			},
+			icon: MessageStripTemplateContext.getIcon(state),
+		};
+	}
+
+	static getIcon(state) {
+		const mappings = {
+			"Positive": "sap-icon://message-success",
```

Comment:
move declaration outside to reuse it every time as its static, you don`t need to create object every time the component re-renders.